crates/hyperswitch_connectors/src/connectors/elavon/transformers.rs

```diff
@@ -1,228 +1,587 @@
-            ElavonPaymentStatus::Succeeded => Self::Charged,
-            ElavonPaymentStatus::Failed => Self::Failure,
-            ElavonPaymentStatus::Processing => Self::Authorizing,
-        }
-    }
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+enum SslIssuerResponse {
+    #[serde(rename = "00")]
+    Success,
+    #[serde(rename = "85")]
+    NoDeclineReason,
+    #[serde(rename = "10")]
+    PartialSuccess10,
+    #[serde(rename = "11")]
+    PartialSuccess11,
+    #[serde(other)]
+    Unknown,
+}
+#[derive(Debug, Clone, Serialize, Deserialize)]
+enum SslResult {
+    #[serde(rename = "0")]
+    ImportedBatchFile,
+    #[serde(other)]
+    DeclineOrUnauthorized,
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
-pub struct ElavonPaymentsResponse {
-    status: ElavonPaymentStatus,
-    id: String,
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum ElavonPaymentsResponse {
+    #[serde(rename = "txn")]
+    Success(PaymentResponse),
+    #[serde(rename = "txn")]
+    Error(ElavonErrorResponse),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ElavonErrorResponse {
+    error_code: String,
+    error_message: String,
+    error_name: String,
+}
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct PaymentResponse {
+    ssl_issuer_response: Option<SslIssuerResponse>,
+    ssl_result: SslResult,
+    ssl_txn_id: String,
+    ssl_result_message: Option<String>,
 }
 
-impl<F, T> TryFrom<ResponseRouterData<F, ElavonPaymentsResponse, T, PaymentsResponseData>>
-    for RouterData<F, T, PaymentsResponseData>
+impl<F>
+    TryFrom<
+        ResponseRouterData<F, ElavonPaymentsResponse, PaymentsAuthorizeData, PaymentsResponseData>,
+    > for RouterData<F, PaymentsAuthorizeData, PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: ResponseRouterData<F, ElavonPaymentsResponse, T, PaymentsResponseData>,
+        item: ResponseRouterData<
+            F,
+            ElavonPaymentsResponse,
+            PaymentsAuthorizeData,
+            PaymentsResponseData,
+        >,
     ) -> Result<Self, Self::Error> {
-        Ok(Self {
-            status: common_enums::AttemptStatus::from(item.response.status),
-            response: Ok(PaymentsResponseData::TransactionResponse {
-                resource_id: ResponseId::ConnectorTransactionId(item.response.id),
-                redirection_data: Box::new(None),
-                mandate_reference: Box::new(None),
-                connector_metadata: None,
-                network_txn_id: None,
-                connector_response_reference_id: None,
-                incremental_authorization_allowed: None,
-                charge_id: None,
+        let status = enums::AttemptStatus::foreign_from((
+            &item.response,
+            item.data.request.is_auto_capture()?,
+        ));
```

Comment:
Please define a function instead of using `ForeignFrom` trait

Reply:
defined